hack/update-codegen.sh

```diff
@@ -69,11 +69,55 @@ ${CODEGEN_PKG}/generate-groups.sh "all" \
 #go build -o defaulter-gen ${CODEGEN_PKG}/cmd/defaulter-gen
 
 echo "Generating defaulters for tensorflow/v1"
-${GOPATH}/bin/defaulter-gen  --input-dirs github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1 -O zz_generated.defaults --output-package github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1 --go-header-file hack/boilerplate/boilerplate.go.txt "$@"
+${GOPATH}/bin/defaulter-gen --input-dirs github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1 \
```

Comment:
Do we want to extract version for the code-generator from [`go.mod`](https://github.com/kubeflow/tf-operator/blob/master/go.mod#L17) to have similar generation in different environment ?

Reply:
Sounds good. I can make the change to build `defaulter-gen` from pkg and use it here.

Reply:
Update: I determine to stick to current solution but leave some comments. 

When we run the scripts, it automatically install binaries in GOPATH/bin.
https://github.com/kubeflow/tf-operator/blob/7604da5661318fb7d27b65d906f28ae5d382547c/hack/update-codegen.sh#L52


https://github.com/kubernetes/code-generator/blob/ee3ba0fad71e3afe882f946036856ada0a0fc603/generate-groups.sh#L52-L53

Reply:
https://github.com/kubeflow/tf-operator/blob/7604da5661318fb7d27b65d906f28ae5d382547c/hack/update-codegen.sh#L60

Reply:
Make sense, @Jeffwan for `openapi-gen ` it works with the same way ?

Reply:
@andreyvelich em. nice catch. actually openapi-gen comes from https://github.com/kubernetes/kube-openapi and seems we didn't build binary locally. That means user doesn't have it in path will fail the scripts. Let me add it.

Reply:
I updates the scripts to build and use openapi-gen from go.mod version